plugins/outputs/postgresql/columns.go

```diff
@@ -13,14 +12,24 @@ const (
 	jsonColumnDataType   = PgJSONb
 )
 
-var timeColumn = utils.Column{Name: timeColumnName, Type: timeColumnDataType, Role: utils.TimeColType}
-var tagIDColumn = utils.Column{Name: tagIDColumnName, Type: tagIDColumnDataType, Role: utils.TagsIDColType}
-var fieldsJSONColumn = utils.Column{Name: fieldsJSONColumnName, Type: jsonColumnDataType, Role: utils.FieldColType}
-var tagsJSONColumn = utils.Column{Name: tagsJSONColumnName, Type: jsonColumnDataType, Role: utils.TagColType}
+var (
+	timeColumnName   = "time"
+	timeColumn       = utils.Column{Name: timeColumnName, Type: timeColumnDataType, Role: utils.TimeColType}
+	tagIDColumn      = utils.Column{Name: tagIDColumnName, Type: tagIDColumnDataType, Role: utils.TagsIDColType}
+	fieldsJSONColumn = utils.Column{Name: fieldsJSONColumnName, Type: jsonColumnDataType, Role: utils.FieldColType}
+	tagsJSONColumn   = utils.Column{Name: tagsJSONColumnName, Type: jsonColumnDataType, Role: utils.TagColType}
+)
 
 func (p *Postgresql) columnFromTag(key string, value interface{}) utils.Column {
 	return utils.Column{Name: key, Type: p.derivePgDatatype(value), Role: utils.TagColType}
 }
 func (p *Postgresql) columnFromField(key string, value interface{}) utils.Column {
 	return utils.Column{Name: key, Type: p.derivePgDatatype(value), Role: utils.FieldColType}
 }
+
+func (p *Postgresql) initTimeColumn() {
+	if p.TimestampColumnName != "" {
+		timeColumnName = p.TimestampColumnName
+		timeColumn = utils.Column{Name: timeColumnName, Type: timeColumnDataType, Role: utils.TimeColType}
```

Comment:
my $0.02, I would do away with `timeColumnName`. If it's variable, then you have to keep it in sync with everything that copies it (like you're doing here). It'd likely be cleaner to get rid of it and do `timeColumn.Name = p.TimestampColumnName`. On top of that, if you make `p.TimestampColumName` default to `time`, then you can get rid of the conditional. I think it would also be more consistent to keep all the defaults in one place, rather than having this special case.

Reply:
@phemmer can you please check if the update is what you had in mind!? I also adapted the other column types to also use `column.Name`...